### PR TITLE
Feature/aria label box plot

### PR DESCRIPTION
**Pull request by ljones87**
This PR adds `aria-label` and `tabIndex` props to the `Whisker`, `LineSegment`, and `Border` components of `VictoryBoxPlot`  :

ariaLabels and tabIndex's will need to be added on a component basis

I added them to the following data points in the demo: 
max
q3
median
q1
min

Can verify via `yarn start` or `yarn start-ts` where the box plot has been added to the accessibility demo page.

min
q1box
median
q3box
max

ToDo: 
- [x] tests

### CODE REVIEW

File: demo/js/components/accessibility-demo.js [hunk 1]
```diff
@@ -1,45 +1,111 @@
...
+                 * _median: 5.5
+                 * _min: 3
+                 * _q1: 4.5
+                 * _q3: 6.75
+                 * _x: 3
+                 * _y: 3
+                 */
+                maxComponent={
+                  <Whisker
+                    ariaLabel={({ datum }) => `${datum.x} max is ${datum._max}`}
```
Comment (boygirl) [inline demo/js/components/accessibility-demo.js:76]: It might be useful to have this example include `tabIndex` / `ariaLabel` props on both `VictoryBoxPlot` itself and on some of the components it renders.

**Review by boygirl — COMMENTED** [review]
After playing with the box plot demo, I'm pretty on the fence about whether we should add the `tabIndex` and `ariaLabel` props to components like `VictoryBoxPlot` and `VictoryBar` vs. adding them only to the primitive components. 

It was really cool to be able to tab through each element in the demo when you added a different `tabIndex` and `ariaLabel` to each type of dataComponent, but I'm not sure the default behavior you get when you add `tabIndex` and `ariaLabel` to `VictoryBoxPlot` is desirable. It's pretty convenient to be able to make all of those elements tabbable without having to add a `tabIndex` to each primitive component. On the other hand, doing so includes it on all the labels as well, which might feel redundant. There also isn't a way to be specific enough with the `ariaLabel` when it's added to `VictoryBoxPlot` rather than on the primitive components, since the primitive components don't know whether they are min whiskers or max whiskers, for example. What do you think?

**Comment by ljones87** [comment]
@boygirl I definitely agree in the two I have PR'd so far since they render a component for each data point. 
However, yesterday I was having issue with area and line where the primitives do not have access to datum, which I'm sure can be fixed, but also `VictoryLine`s primitive `Curve` is only rendered once which makes sense as it is a continuous line. Its cases like this as well as `VictoryArea` I'm not sure the best way to handle. 
On `VictoryLine`, the `labels` have access to each data point, so thought in that case we would apply the `tabsIndex` and `ariaLabel` the same way. 
All above "issues" could just be my naivety with the library/data components. I've only spent an hour to two digging into area and line so there may be answers when I spend more time on them today.

**Comment by boygirl** [comment]
@ljones87 `VictoryLine` and `VictoryArea` render single, continuous paths for the entire data set, so the `Curve` and `Area` primtive components have no concept of `datum` since they never correspond to individual points. There's not really a way around that. An aria label on a line would describe the whole data set. Points could be aria-labelled individually on the label components as you suggest.

**Comment by ljones87** [comment]
So for VictoryLine/Area etc, the only way to add aria-labels would be via the label component, am I understanding that correctly? Since those primitives cannot support `ariaLabel` and `tabIndex`

**Comment by boygirl** [comment]
No, the `Curve` and `Area` primitives should support `ariaLabel` and `tabIndex`. They could be used to label the entire series. It just wont be possible to label individual points on these primities since they render only one element for the entire dataset.

**Comment by ljones87** [comment]
ok can do! If the user wants the data points aria-labelled then they'd need to use the labels/labelComponent? aka I still will not be adding a top level tabIndex/ariaLabel to say `VictoryLine`?

**Comment by boygirl** [comment]
Yep, perfect! Thank you!

File: packages/victory-bar/src/index.d.ts [hunk 2]
```diff
@@ -19,7 +19,6 @@ export interface VictoryBarProps
     VictoryDatableProps,
     VictoryMultiLabelableProps {
   alignment?: VictoryBarAlignmentType;
-  ariaLabel?: StringOrCallback;
```
Comment (ljones87) [inline packages/victory-bar/src/index.d.ts:22]: Removed these types from `VictoryBar`, although the props if provided will still apply, just gives a ts error in the demo. There doesn't appear to be a way to prevent the ariaLabel from applying to `VictoryBar` separate from `Bar`

File: packages/victory-core/src/index.js [hunk 3]
```diff
@@ -9,9 +9,15 @@ export { default as Portal } from "./victory-portal/portal";
...
 export { default as Border, default as Box } from "./victory-primitives/border";
+export { default as Circle } from "./victory-primitives/circle";
 export { default as ClipPath } from "./victory-primitives/clip-path";
+export { default as Line } from "./victory-primitives/line";
 export { default as LineSegment } from "./victory-primitives/line-segment";
+export { default as Path } from "./victory-primitives/path";
 export { default as Point } from "./victory-primitives/point";
+export { default as Rect } from "./victory-primitives/rect";
+export { default as Text } from "./victory-primitives/text";
+export { default as TSpan } from "./victory-primitives/tspan";
```
Comment (ljones87) [inline packages/victory-core/src/index.js:20]: just grouped all the primitives together as I was figuring out what primitives were a part of box plot, happy to revert this change. I haven't investigated what the color coding represents on these exports

File: packages/victory-box-plot/src/helper-methods.js [hunk 4]
```diff
@@ -206,12 +206,23 @@ const getCalculatedValues = (props) => {
 
 // eslint-disable-next-line complexity
 const getWhiskerProps = (props, type) => {
-  const { horizontal, style, boxWidth, whiskerWidth, datum, scale, index } = props;
+  const {
+    ariaLabel,
```
Comment (boygirl) [inline packages/victory-box-plot/src/helper-methods.js:210]: `ariaLabel` and `tabIndex` props shouldn't be passed from `VictoryBoxPlot` to its primitives. Only the primitives need to know about these props. `VictoryBoxPlot`, `VictoryBar` etc shouldn't use them

File: demo/js/components/accessibility-demo.js [hunk 5]
```diff
@@ -1,45 +1,110 @@
...
+            <h3> Bar chart</h3>
+            <VictoryChart domainPadding={{ x: 40, y: 40 }}>
+              <VictoryBar
+                data={[
+                  { x: "A", y: 1 },
+                  { x: "B", y: 3 },
+                  { x: "C", y: 5 },
+                  { x: "D", y: 7 }
+                ]}
+                ariaLabel={({ datum }) => `bar-value-${datum.x}`}
```
Comment (boygirl) [inline demo/js/components/accessibility-demo.js:42]: applying `ariaLabel` and `tabIndex` to `VictoryBar` shouldn't work any more after the API change we discussed, so this demo should be changed.

Comment (ljones87) [reply demo/js/components/accessibility-demo.js:42]: aaah, my bad, I misunderstood how the props in helpers were applying. Box plot and bar have been updated as have the tests

**Comment by ljones87** [comment]
@boygirl do you have a preference on commits/do you prefer them squashed to a single feature commit message?

File: packages/victory-core/src/index.d.ts [hunk 6]
```diff
@@ -88,6 +88,7 @@ export interface CallbackArgs {
   active: boolean;
   datum: any;
   horizontal: boolean;
+  index: number;
```
Comment (boygirl) [inline packages/victory-core/src/index.d.ts:91]: actually I think this needs to be `number | string`

**Comment by boygirl** [comment]
@ljones87 This is looking good. I left a few small comments, but aside from those and some clean up, I think it's ready to merge.

**Comment by boygirl** [comment]
> @boygirl do you have a preference on commits/do you prefer them squashed to a single feature commit message?

I don't have a preference

File: packages/victory-core/src/index.d.ts (hunk 6 above)
Comment (ljones87) [reply packages/victory-core/src/index.d.ts:91]: I know it ends up as a string in the dom, but don't we always want the input to be a number?

File: test/client/spec/victory-bar/victory-bar.spec.js [hunk 7]
```diff
@@ -200,5 +200,55 @@ describe("components/victory-bar", () => {
...
+      wrapper.find("path").forEach((p, i) => {
+        const index = p.prop("index");
+        const tabIndex = p.prop("tabIndex");
+        expect(tabIndex).to.equal(index + 1);
+        const ariaLabel = p.prop("aria-label");
+        expect(ariaLabel).to.equal(`x: ${data[i].x}`);
+      });
+    });
+
+    it("does not apply tabIndex or aria label via the parent", () => {
```
[the remark this answers is not in the document]
Comment (ljones87) [reply test/client/spec/victory-bar/victory-bar.spec.js:223]: ok cool, figured for "documentation" purposes but that is covered by TS and if/when we update the docs

File: packages/victory-core/src/index.d.ts (hunk 6 above)
Comment (boygirl) [reply packages/victory-core/src/index.d.ts:91]: This prop is set by components like `VictoryArea`. In the case where they only render a single component for the entire dataset they set the value "all"

Comment (ljones87) [reply packages/victory-core/src/index.d.ts]: aaah, yeah thats right, fixed 👍

File: demo/js/components/accessibility-demo.js (hunk 5 above)
Comment (boygirl) [inline demo/js/components/accessibility-demo.js:42]: `ariaLabel` and `tabIndex` should be on the `Bar` `dataComponent` in this example.

**Comment by boygirl** [comment]
@ljones87 I added one minor comment on one of your demo examples. Otherwise this is approved.
